src/components/notifications/notifications.tsx

```diff
@@ -1,86 +1,121 @@
+  errorOrMessage?: Error | string;
+  sendToSentry?: boolean;
+} & ToastProps;
 
-interface ToastProps {
-  message: ReactNode;
-  url?: string;
-}
+const GenericIcon = (props: { type: Partial<ToastOptions['type']> }) => {
+  const icons = {
+    info: <InfoCircleFillIcon className={classNames(styles.icon, styles.info)} />,
+    success: <CheckCircleFillIcon className={classNames(styles.icon, styles.success)} />,
+    error: <ExclamationTriangleFillIcon className={classNames(styles.icon, styles.error)} />,
+    warning: <WarningCircleFillIcon className={classNames(styles.icon, styles.warning)} />,
+  };
 
-interface ErrorToastProps extends ToastProps {
-  errorOrMessage: Error | string;
-  sendToSentry?: boolean;
-}
+  return icons[props.type as keyof typeof icons];
+};
 
-interface ToastPropsWithType extends ToastProps {
-  type: 'info' | 'success' | 'warning' | 'error';
-}
+const CustomToast = (props: ToastProps & ToastOptions) => {
+  const { customAction, message, type, url, autoClose } = props;
 
-const CustomToast = ({ message, type, url }: ToastPropsWithType) => {
   return (
-    <div className={classNames(styles.notificationBody, { [styles.url]: url })}>
-      <img src={`/${type}.svg`} alt={`${type} icon`} />
+    <div className={styles.notification}>
+      <GenericIcon type={type} />
+
       <div className={styles.notificationContent}>
-        <div>{message}</div>
+        <span className={styles.notificationMessage}>{message}</span>
+
         {url && (
-          <div className={styles.notificationUrl}>
-            <NotificationLinkButton href={url}>View transaction</NotificationLinkButton>
+          <div className={styles.transactionButtonContainer}>
+            <Button type="text-gray" size="xs" className={styles.transactionButton} href={url}>
+              <span>View transaction</span>
+              <ExternalLinkIcon />
+            </Button>
           </div>
         )}
+
+        {customAction}
       </div>
+
+      <div
+        className={classNames(styles.progressBarBackground, {
+          [styles[`${type}`]]: autoClose === false,
+        })}
+      />
     </div>
   );
 };
 
 // https://fkhadra.github.io/react-toastify/api/toast
 const BASE_OPTIONS: ToastOptions = {
   transition: Slide,
-  closeButton: CloseButton,
-  hideProgressBar: false,
+  closeButton: (props) => (
+    <button className={styles.closeButton} onClick={props.closeToast}>
+      <CrossIcon />
+      <span className="sr-only">Close</span>
+    </button>
+  ),
 };
 
 // NOTE: toasts are throttled to prevent duplicate notifications being displayed.
 // This can occur due to callbacks being fired multiple times in quick succession
 export const info = throttle(
   (props: ToastProps, overrides?: ToastOptions) => {
-    return toast.info(<CustomToast {...props} type="info" />, { ...BASE_OPTIONS, ...overrides });
+    return toast.info(<CustomToast {...props} {...overrides} type="info" />, { ...BASE_OPTIONS, ...overrides });
```

Comment:
Hmm any reason for change in spreading the `overrides`/options as props?

Reply:
Yeah, I've done it so that `CustomToast` can get the `autoClose` prop, which is passed as a `ToastOptions` (second argument), not as a `ToastProps` (first argument). See [here](https://github.com/api3dao/api3-dao-dashboard/blob/3c08e1c0139347dfd2773e33b269fc702a27fd21/src/app.tsx#L126). We need to pass it that way so that it ends up as an override.

And because `CustomToast` already expected `ToastOptions`, I felt like spreading is the cleanest approach. But we can also do something like this `{...props} autoClose={overrides.autoClose}`, either way would work. If you see a cleaner way, please shoot 🙏 

<img width="767" alt="image" src="https://github.com/user-attachments/assets/67b59e6d-fe81-4362-a088-c9f133919a53" />